04week/mastermind.js

```diff
@@ -17,31 +17,56 @@ function printBoard() {
   }
 }
 
-function generateSolution() {
-  for (let i = 0; i < 4; i++) {
-    const randomIndex = getRandomInt(0, letters.length);
-    solution += letters[randomIndex];
-  }
-}
+// function generateSolution() { //creates the randomized solution 
+//   for (let i = 0; i < 4; i++) {
+//     const randomIndex = getRandomInt(0, letters.length);
+//     solution += letters[randomIndex];
+//   }
+// }
+
+// function getRandomInt(min, max) { //creates rando interger to use in generate soln?
+//   return Math.floor(Math.random() * (max - min)) + min;
+// }
 
-function getRandomInt(min, max) {
-  return Math.floor(Math.random() * (max - min)) + min;
-}
 
-function generateHint() {
-  // your code here
+function generateHint(guess) {
+  let countExact = 0; 
+  let countLetter = 0; 
+  // could i have used: for (let i of guess){} ? //
+  for (let i= 0; i<= guess.length; i++) {
+    for (let j= 0; j< 4; j++) {//for each [i] of array, compare every [j] of array  
+      if(guess[i] == solution[i]){ //for exact match
+        countExact ++;  //add to count 
+        solution[i] = 'null';//take it off soln [] so it won't be double counted
+        //add to board [] <- has all the exact matches    //<- the test seems to use board.length for hints?
+
+        
+      }else if(guess[i] == solution[j]){//for letter only match
+        countLetter++;  //add to count
+        solution[j] = 'null';//take off solution 
+        // board.push(solution[j]);//add to board [] <- now has all the exact & letter only matches, including doubles, in order:exact,letter
+      };
+      return `${countExact} - ${countLetter}`;
+    };
+  };
 }
 
-function mastermind(guess) {
-  solution = 'abcd'; // Comment this out to generate a random solution
-  // your code here
+
+function mastermind(guess) { 
```

Comment:
I was not able to run the program and play the game, and the tests did not pass.